src/sdk/config_fs.js

```diff
@@ -468,6 +496,46 @@ class ConfigFS {
         return nb_native().fs.stat(this.fs_context, path_for_account_or_user_config_file);
     }
 
+    /**
+     * stat_account_config_file_by_identity will return the stat output on account config file by id or by account name
+     * @param {string} id
+     * @param {string} account_name
+     * @returns {Promise<nb.NativeFSStats>}
+     */
+    async stat_account_config_file_by_identity(id, account_name) {
+        const identity_path = this.get_identity_path_by_id(id);
+        let stat;
+        try {
+            const stat_by_identity_path = await nb_native().fs.stat(this.fs_context, identity_path);
+            stat = stat_by_identity_path;
+        } catch (err_stat_by_identity_path) {
+            if (err_stat_by_identity_path.code === 'ENOENT') {
+                dbg.warn('get_identity_by_id_and_stat_file: could not stat by identity ID will try to stat by account name');
+                const account_name_new_path = this.get_account_path_by_name(account_name);
+                try {
+                    const stat_by_account_name = await nb_native().fs.stat(this.fs_context, account_name_new_path);
+                    stat = stat_by_account_name;
+                } catch (err_stat_by_account_name_new_path) {
+                    if (err_stat_by_identity_path.code === 'ENOENT') {
+                        dbg.warn('get_identity_by_id_and_stat_file: could not stat by by account name (new accounts path)');
+                        const account_name_old_path = this._get_old_account_path_by_name(account_name);
+                        try {
+                            const stat_by_account_name_old = await nb_native().fs.stat(this.fs_context, account_name_old_path);
+                            stat = stat_by_account_name_old;
+                        } catch (err_stat_by_account_name_old_path) {
+                            dbg.warn('get_identity_by_id_and_stat_file: could not stat by by account name (old accounts path)');
+                            // eslint-disable-next-line max-depth
+                            const error_to_throw = new Error(`Could not stat identity by id ${id} or by account name ${account_name}`);
+                            error_to_throw.code = 'ENOENT';
+                            throw error_to_throw;
```

Comment:
this seems to be the only place we actually throw. in all the previous try catch there are no other cases that needs to throw. say when the error isn't ENOENT

Reply:
It is the last way that we have to stat the file after we tried all the other ways, so we will throw the error.

Reply:
right, but you ignore all errors that are not ENOENT along the way. I don't think any of them should be raised. but still if you refer to ENOENT specifically might as well handle the others as well.